watsononlinestore/watson_online_store.py

```diff
@@ -303,6 +305,19 @@ def get_discovery_response(self, input_text):
             query_options={'query': input_text, 'count': DISCOVERY_QUERY_COUNT}
         )
 
+        # Watson discovery assigns a confidence level to each result.
+        # Based on data mix, we can assign a minimum tolerance value in an
+        # attempt to filter out the "weakest" results.
+        if 'results' in discovery_response and self.discovery_score_filter:
+            filtered_results = []
+            for result in discovery_response['results']:
+                if 'score' in result:
+                    if result['score'] >= self.discovery_score_filter:
+                        filtered_results.append(result)
+
+            discovery_response['matching_results'] = len(filtered_results)
```

Comment:
num_  was more obvious. I might expect matching_results to actuall hold the results.
Most likely it would be better to leave it out and let some other code call len() when needed -- but not sure where that is right now. Don't break it.